### DISCUSSION TITLE
Inconsistent Code

### DISCUSSION THREAD

**Discussion by rmccu** [jongpie/NebulaLogger#377] [Q&A] [answered]
https://github.com/jongpie/NebulaLogger/blob/3b04138cdb22211796870ef49d8b3773465641e9/nebula-logger/core/main/log-management/classes/LogEntryEventHandler.cls#L162

Why does it use this format (setting the attributes inside of the LogEntry__c() constructor in the insertLogEntries() method, but in the upsertLogs() it creates a new instance of Log__c and then sets each attribute. Seems inconsistent. If there is a reason for this, there should be a comment explaining it.

**Comment by jongpie** [comment, marked as answer]
@rmccu the commit you're referencing, [3b04138](https://github.com/jongpie/NebulaLogger/commit/3b04138cdb22211796870ef49d8b3773465641e9), is 4-5 months old (v4.7.1). In the latest version, [v4.8.3](https://github.com/jongpie/NebulaLogger/blob/v4.8.3/nebula-logger/core/main/log-management/classes/LogEntryEventHandler.cls#L222), the constructors are now used for both `Log__c` and `LogEntry__c` to set most fields.

**Reply by rmccu** [reply]
Not sure how I found the old version. I think I just searched and it took me to the old version. I didn't think to look at a newer version or "main".
